### DISCUSSION TITLE
Has anyone played with SeaweedFS and Talos? Other small-scale storage solutions?

### DISCUSSION THREAD

**Discussion by alexpdp7** [siderolabs/talos#8182] [Q&A]
My use case is two single-node Talos clusters. I know single-node k8s does not make a ton of sense, but this is personal infra. I just use k8s as a small PaaS to deploy some applications. Also for learning purposes and because increasingly, more software is easier to deploy on k8s.

For now, I've been running a storageless cluster, which has been fine for my purposes. However, I want to run a piece of software that supports object storage ([Takahe](https://jointakahe.org/)). So I was looking at setting up Minio or some other "easy" object storage outside the cluster.

But SeaweedFS caught my attention. I got a POC very quickly (single command), and I read some [encouraging article](https://github.com/internetarchive/sandcrawler/blob/master/proposals/2020_seaweed_s3.md)s. It apparently can do object storage and provide PVs to Kubernetes clusters.

Anyone using this with Talos? I would run it alongside my Talos clusters, in Proxmox. Talos is a VM, while I would run this as an LXC container.

**Comment by bitsondatadev** [comment]
@alexpdp7 did you ever try this? I am looking at this now with a similar use case.

**Reply by alexpdp7** [reply]
Well, I'm running SeaweedFS for Takahe, so far no issues, but the load is very low. But I only use it to provide object storage to applications. The applications are running in Talos, and SeaweedFS is running outside of Talos, but really I haven't done anything Talos-specific (like using SeaweedFS to provide storage for the cluster, or run SeaweedFS in Talos).

(As a small plug, I run my own COPR for EL9 here <https://copr.fedorainfracloud.org/coprs/koalillo/seaweedfs/builds/>.)

**Reply by bitsondatadev** [reply]
That's great! The way you are running SeaweedFS services independent of Talos is exactly how I would want to do it though I am tempted to give the [seaweedfs-operator](https://github.com/seaweedfs/seaweedfs-operator) a shot. I likely wouldn't want the SeaweedFS services running in k8s to depend on itself as the filesystem. I'd rather them just use local since it avoids the circular dependency.

I am setting up a homelab that can run some services for both personal workloads and scale up for business uses over time. This service needs to support both block and object storage patterns. My main concern is to scale/manage services and storage independently (like any cloud provider) but do it with various sizes of workloads, most likely smaller to begin with which makes Ceph less desirable. The key for me is flexibility of use cases over dominance over the larger use cases.

Thanks for sharing your COPRs. I have very little experience with Fedora/RHEL distros. If I don't end up using the kubernetes operator, do you suggest Fedora for this use case or are you just familiar with it?

Thanks!

**Reply by alexpdp7** [reply]
> Thanks for sharing your COPRs. I have very little experience with Fedora/RHEL distros. If I don't end up using the kubernetes operator, do you suggest Fedora for this use case or are you just familiar with it?

Oh, the COPR only provides packages for RHEL 9 and clones. Fedora is great, but I prefer to run distributions with longer support cycles for my infrastructure.

**Reply by bitsondatadev** [reply]
Ah, is RHEL pretty accessible to a Debian-based user?

I used it for a couple years in industry and there were just so many nuances I wasn't used to since I never used it outside of work. Which taxonomies of use cases classes you recommend RHEL and/or enterprise support for?

1. Personal homelab setup
   1. high experimentation
   1. low security
   1. low-high latency/bandwidth
   1. random arbitrary downtime expected
1. Personal homelab servicing friends and family
   1. med-low experimentation
   1. med-high security with many family members data
   1. low-high latency/bandwidth
   1. occasional planned downtime with higher risk of unplanned downtime
1. Small business for friends/family/customer data
   1. low experimentation (likely some testing environment and cicd)
  1. high security
  1. med-high latency/bandwidth
  1. occasional planned downtime with lower risk of unplanned downtime

My hope is that with new automation frameworks and decentralized and modular services, you enable smaller businesses to handle most of their own enterprise use cases or male it possible for more distributed community based support.

I do still think the most solid way is centralized support but I think that trend is to customize and localize pain points where more complexity is needed. In this case, grenade launchers like RHEL might be overkill if you have a small business, but always depends on specific I guess.

**Reply by alexpdp7** [reply]
I suspect this is veering off-topic; I'll answer briefly, feel free to email me if you want further discussion.

I use RHEL clones out of inertia and minor factors, but I'd always default to LTS distros (Ubuntu LTS, Debian Stable, RHEL clones, etc.) because on rolling distros, any update could require manual intervention to update configuration. With an LTS distro, I know that if I configure a system, it should not require configuration updates for a couple of years. I do not use paid enterprise support.

**Reply by bitsondatadev** [reply]
Thanks Alex that was quite helpful and mostly what I needed! Thanks for offering to chat offline!

I'll report back once I have Talks and SeaweedFS set up.

I think I may give that k8s operator a go on Talos a go.

Edit: I'm running RPi5 and ran into this issue: #7978. I'm gonna use k3s for now but will come back to talos once rpi5 is ready.

**Comment by saolof** [comment]
You may also be interested in [garage](https://garagehq.deuxfleurs.fr/documentation/cookbook/kubernetes/) if your main usecase is ease of self hosting. Due to not having or supporting anything that can lock up it needs less orchestration and you can get away with just using a statefulset.

**Comment by GioPat** [comment]
I am also using Talos as single node for a home server relying on ZFS, for backup needs I ended up looking for SeaweedFS which seems a promising alternative. I wonder if it should be managed as a Talos plugin or as an k8s deployment overlay.
